docs/ossm/ossm2-migration/cluster-wide/README.md

```diff
@@ -0,0 +1,96 @@
+
+### Prerequisites
+- you have completed all the steps in the [pre-migration checklist](../README.md#pre-migration-checklist)
+- you have read [OpenShift Service Mesh 2 vs. OpenShift Service Mesh 3](../../ossm2-vs-ossm3.md)
+- you have verified that your 2.6 `ServiceMeshControlPlane` is using `ClusterWide` mode
+- Red Hat OpenShift Service Mesh 3 operator is installed
+- `IstioCNI` is installed
+- `istioctl` is installed
+
+### Procedure
+In this example, we'll be using the [bookinfo demo](https://raw.githubusercontent.com/Maistra/istio/maistra-2.6/samples/bookinfo/platform/kube/bookinfo.yaml) but you can follow these same steps with your own workloads.
+
+TODO: should we add a note about `minReplicas` for workload deployments?
+TODO: include also ingress gateway deployment migration?
+
+#### Plan the migration
+There will be two cluster wide Istio control planes running during the migration process so it's necessary to plan the migration steps in advance in order to avoid possible conflicts between the two control planes.
+
+There is a few conditions which must be assured to achieve the seamless migration:
+- both control planes must share the same root certificate
+
+  This can be achieved simply by installing the 3.0 control plane to the same namespace as 2.6 control plane
+- both control planes have access to all migrated namespaces and namespaces running addons (e.g. otel collector)
+  1. verify there are no Network Policies blocking the traffic
+  1. `discoverySelectors`  defined in 3.0 `Istio` resource must match migrated namespaces. See [Scoping the service mesh with DiscoverySelectors](../../create-mesh/README.md)
+- only one control plane will try to inject a side car
+
+To achieve last condition, it's necessary to understand how the injection works. Please see [Installing the Sidecar](../../injection/README.md) for details.
+> [!CAUTION]
+> In case you decide to use a `default` name for 3.0 `Istio` resource with `InPlace` upgrade strategy, 3.0 control plane will try to inject side cars to all pods in namespaces with `istio-injection=enabled` label and all pods with `sidecar.istio.io/inject="true"` label after next restart of the workloads. Without using `maistra.io/ignore-namespace: "true"` label, it will result in a conflict with 2.6 control plane injector.
+
+In case you are using `istio-injection=enabled` label on your data plane namespaces and planning to keep it, you should first read revision tag documentation. TODO: add link. TODO: describe cases where the migration will work with `istio-injection=enabled` (default revision, default revision tag)
+
+
+#### Install OpenShift Service Mesh 3
+
+1. Create your `Istio` resource.
+
+    Here we are not using any `discoverySelectors` so the control plane will have access to all namespaces. In case you want to define `discoverySelectors`, keep in mind that all data plane namespaces you are planning to migrate must be matched.
```

Comment:
I don't think you need to keep mentioning `discoverySelectors`. Or else it can be a separate guide? Having a guide that is dedicated to single tenant cluster wide could help keep it more focused.

Reply:
I also wonder too, if, rather than giving specific instructions to create a new Istio resource here, we could direct them to the OSSM installation instructions, which covers discoverySelectors. I notice that the `IstioCNI` resource being installed is a pre-req, but that is only discussed in the main installation page.  Or, as we have the running OSSM 2 and OSSM 3 side by side instructions, if we could direct people toward those and then use that as the starting point? Just trying to reduce duplication...

Reply:
The goal of mentioning `discoverySelectors` here is to avoid situations where users incorrectly define `discoverySelectors` which will prevent access to some 2.6 namespaces which they plan to migrate. It would cause a problem where 3.0 internal service registry would be different from 2.6 internal service registry meaning that some communication would break during the migration. This problem should be mentioned in this migration guide very clearly I believe as it's not obvious.